src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/cancel_interaction_request.cc

```diff
@@ -0,0 +1,129 @@
+distribution.
+
+Neither the name of the the copyright holders nor the names of their
+contributors may be used to endorse or promote products derived from this
+software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#include "sdl_rpc_plugin/commands/mobile/cancel_interaction_request.h"
+#include "application_manager/application_impl.h"
+#include "application_manager/message_helper.h"
+#include "interfaces/MOBILE_API.h"
+#include "utils/helpers.h"
+
+namespace sdl_rpc_plugin {
+using namespace application_manager;
+namespace commands {
+
+CancelInteractionRequest::CancelInteractionRequest(
+    const application_manager::commands::MessageSharedPtr& message,
+    ApplicationManager& application_manager,
+    app_mngr::rpc_service::RPCService& rpc_service,
+    app_mngr::HMICapabilities& hmi_capabilities,
+    policy::PolicyHandlerInterface& policy_handler)
+    : CommandRequestImpl(message,
+                         application_manager,
+                         rpc_service,
+                         hmi_capabilities,
+                         policy_handler) {}
+
+CancelInteractionRequest::~CancelInteractionRequest() {}
+
+void CancelInteractionRequest::Run() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  auto app = application_manager_.application(connection_key());
+
+  if (!app || app.use_count() == 0) {
+    LOG4CXX_ERROR(logger_, "Application does not exist");
+    SendResponse(false, mobile_apis::Result::APPLICATION_NOT_REGISTERED);
+    return;
+  }
+
+  auto function_id = (*message_)[strings::msg_params][strings::func_id].asInt();
+  auto cancel_id = (*message_)[strings::msg_params][strings::cancel_id].asInt();
+
+  if (function_id != FunctionID::PerformInteractionID &&
+      function_id != FunctionID::AlertID &&
+      function_id != FunctionID::ScrollableMessageID &&
+      function_id != FunctionID::SliderID) {
+    LOG4CXX_ERROR(logger_, "Bad function ID" << function_id);
+    SendResponse(false, mobile_apis::Result::INVALID_ID);
+    return;
+  }
+
+  smart_objects::SmartObject msg_params;
+  msg_params[strings::app_id] = connection_key();
+  msg_params[strings::func_id] = function_id;
+  msg_params[strings::cancel_id] = cancel_id;
```

Comment:
Remove `cancel_id` variable since it isn't used elsewhere, change this to:
```
if ((*message_)[strings::msg_params].keyExists(strings::cancel_id)) {
  msg_params[strings::cancel_id] = (*message_)[strings::msg_params][strings::cancel_id].asInt();
}
```

This will omit the parameter if it isn't present in the original message, rather than setting it to `-1`.

Reply:
updated everywhere in  https://github.com/smartdevicelink/sdl_core/pull/2963/commits/5fd5f1d7de0d0fa0eb06a66eb9350d5485544ae8